src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -132,9 +132,10 @@ private void ThrowUnableToRunError(Project project)
 
         private void Initialize()
         {
-            if (string.IsNullOrWhiteSpace(Project))
+            var isProjectSet = !string.IsNullOrWhiteSpace(Project);
```

Comment:
I agree that this felt way too hacky.
Just extracted the find logic to a method and it feels more easily understandable now.